lms/lmsweb/tools/registration.py

```diff
@@ -129,6 +167,22 @@ def _build_user_text(user: UserToCreate) -> str:
         return msg
 
 
+def generate_confirmation_token(email: str) -> str:
+    return SERIALIZER.dumps(email, salt='email-confirmation')
```

Comment:
This is problematic, because people that familiar with the source code will be able to confirm accounts which they don't own the email for.

We should generate a random hash using `secrets.random` and save it in the database for each user under `activation_code`, or use some other random generated hash that is valid just for emails.